### PR TITLE
Support convert from UpperLeft, Center in BoundingBoxUtils

### PR DESCRIPTION
**Pull request by am15h**

### CODE REVIEW

File: tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java [hunk 1]
```diff
@@ -197,6 +197,42 @@ private static RectF convertFromBoundaries(
...
 
+  private static RectF convertFromUpperLeft(
+      float[] values, CoordinateType coordinateType, int height, int width) {
+    if (coordinateType == CoordinateType.PIXEL) {
+      float left = values[0];
+      float top = values[1];
+      float w = values[2];
+      float h = values[3];
+
+      return new RectF(
```
Comment (xunkai55) [inline tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java:208]: Please format it correctly based on https://google.github.io/styleguide/javaguide.html, like

```
RectF(left, top, left + w, top + h);
```

So do the rest of code, mostly binary operators.

File: tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java [hunk 2]
```diff
@@ -197,6 +197,42 @@ private static RectF convertFromBoundaries(
...
+        left, top, left+w, top+h);
+    } else {
+      return new RectF(
+        values[0] * width, values[1] * height, (float) width, (float) height);
```
Comment (xunkai55) [inline tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java:212]: I think it should be `width * values[2]` and `height * values[3]`?

File: tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java [hunk 3]
```diff
@@ -197,6 +197,42 @@ private static RectF convertFromBoundaries(
...
+
+      float left = centerX - w/2;
+      float top = centerY - h/2;
+      float right = centerX + w/2;
+      float bottom = centerX + h/2;
+
+      return new RectF(
+        left, top, right, bottom);
+    } else {
+      throw new IllegalArgumentException("BoundingBox.Type Center is not supported with CoordinateType CoordinateType.RATIO");
```
Comment (xunkai55) [inline tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java:232]: IMHO, it looks like able to be supported - perhaps leave a TODO here if you don't want to put all in one PR.

Comment (am15h) [reply tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java:233]: @xunkai55 Thanks for reviewing, I have made other changes, please check.

File: tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java [hunk 4]
```diff
@@ -197,6 +197,43 @@ private static RectF convertFromBoundaries(
...
+      float left = values[0];
+      float top = values[1];
+      float w = values[2];
+      float h = values[3];
+
+      return new RectF(
+        left, top, left + w, top + h);
+    } else {
+      return new RectF(
+        values[0] * width, values[1] * height, width * values[2], height * values[3]);
```
Comment (xunkai55) [inline tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java:212]: Nit: Let's keep the expressions in the same order, like foo * value[bar].

File: tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java [hunk 5]
```diff
@@ -197,6 +197,43 @@ private static RectF convertFromBoundaries(
...
+
+  private static RectF convertFromCenter(
+      float[] values, CoordinateType coordinateType, int height, int width) {
+    if (coordinateType == CoordinateType.PIXEL) {
+      float centerX = values[0];
+      float centerY = values[1];
+      float w = values[2];
+      float h = values[3];
+
+      float left = centerX - w/2;
```
Comment (xunkai55) [inline tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java:224]: Please add spaces around operator `/`.

File: tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java [hunk 6]
```diff
@@ -197,6 +197,43 @@ private static RectF convertFromBoundaries(
...
 
+  private static RectF convertFromUpperLeft(
+      float[] values, CoordinateType coordinateType, int height, int width) {
+    if (coordinateType == CoordinateType.PIXEL) {
+      float left = values[0];
+      float top = values[1];
+      float w = values[2];
+      float h = values[3];
+
+      return new RectF(
```
Comment (xunkai55) [inline tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java:208]: Nit: New line here is not needed.

File: tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java (hunk 4 above)
Comment (xunkai55) [reply tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java:212]: Correct expressions seem to be

values[0] * imageWidth
values[1] * imageHeight
(values[0] + values[2]) * imageWidth
(values[1] + values[3]) * imageHeight

File: tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java [hunk 7]
```diff
@@ -197,6 +197,43 @@ private static RectF convertFromBoundaries(
...
+      float centerY = values[1];
+      float w = values[2];
+      float h = values[3];
+
+      float left = centerX - w/2;
+      float top = centerY - h/2;
+      float right = centerX + w/2;
+      float bottom = centerX + h/2;
+
+      return new RectF(
```
Comment (xunkai55) [inline tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java:229]: Nit: New line is not needed.

File: tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java (hunk 3 above)
Comment (xunkai55) [reply tensorflow_lite_support/java/src/java/org/tensorflow/lite/support/image/BoundingBoxUtil.java:233]: values[2] and [3] represent the relative width and height of the bounding box. So the height and width of the box should be `values[2] * width`, or `imageWidth` as I commented in another thread.

**Comment by am15h** [comment]
Thanks a lot @xunkai55 for reviewing, I have made changes according to your suggestions.

**Review by xunkai55 — APPROVED** [review]

**Comment by xunkai55** [comment]
Thank you @am15h for the contribution. The PR is merged now.

**Pull request merged**